backend/pkg/dbstore/settings.go

```diff
@@ -17,54 +16,122 @@ type Settings struct {
-		stmt.Close()
 	} else {
-		stmt, _ := db.Prepare("INSERT INTO Config (id, grafanaUsername, grafanaPassword, email, emailPassword, datasourceID, emailHost, emailPort) VALUES (?,?,?,?,?,?,?,?)")
-		stmt.Exec("ID", settings.GrafanaUsername, settings.GrafanaPassword, settings.Email, settings.EmailPassword, settings.DatasourceID, settings.EmailHost, settings.EmailPort)
-		stmt.Close()
+		stmt, err := db.Prepare("INSERT INTO Config (id, grafanaUsername, grafanaPassword, email, emailPassword, datasourceID, emailHost, emailPort) VALUES (?,?,?,?,?,?,?,?)")
+		defer stmt.Close()
+		if err != nil {
+			log.DefaultLogger.Error("CreateOrUpdateSettings: db.Prepare()2: ", err.Error())
+			return err
+		}
+
+		_, err = stmt.Exec("ID", settings.GrafanaUsername, settings.GrafanaPassword, settings.Email, settings.EmailPassword, settings.DatasourceID, settings.EmailHost, settings.EmailPort)
+		if err != nil {
+			log.DefaultLogger.Error("CreateOrUpdateSettings: stmt.Exec(): ", err.Error())
+			return err
+		}
+
 	}
-	return true, nil
+	return nil
 }
 
-func (datasource *SQLiteDatasource) GetSettings() *Settings {
-	db, _ := sql.Open("sqlite3", datasource.Path)
+func (datasource *SQLiteDatasource) GetSettings() (*Settings, error) {
+	db, err := sql.Open("sqlite3", datasource.Path)
 	defer db.Close()
+	if err != nil {
+		log.DefaultLogger.Error("GetSettings: sql.Open(): ", err.Error())
+		return nil, err
+	}
 
 	var id, grafanaUsername, grafanaPassword, email, emailPassword, emailHost string
 	var emailPort, datasourceID int
 
-	if datasource.settingsExists() {
+	exists, err := datasource.settingsExists()
+	if err != nil {
+		log.DefaultLogger.Error("GetSettings: settingsExists(): ", err.Error())
+		return nil, err
+	}
 
-		rows, _ := db.Query("SELECT * FROM Config")
+	if exists {
+		rows, err := db.Query("SELECT id, grafanaUsername, grafanaPassword, email, emailPassword, datasourceID, emailHost, emailPort FROM Config")
 		defer rows.Close()
+		if err != nil {
+			log.DefaultLogger.Error("GetSettings: db.Query(): ", err.Error())
+			return nil, err
+		}
+
 		rows.Next()
-		rows.Scan(&id, &grafanaUsername, &grafanaPassword, &email, &emailPassword, &datasourceID, &emailHost, &emailPort)
-		log.DefaultLogger.Warn(id, grafanaUsername, grafanaPassword, email, emailPassword)
-		return &Settings{GrafanaUsername: grafanaUsername, GrafanaPassword: grafanaPassword, Email: email, EmailPassword: emailPassword, DatasourceID: datasourceID, EmailPort: emailPort, EmailHost: emailHost}
+		err = rows.Scan(&id, &grafanaUsername, &grafanaPassword, &email, &emailPassword, &datasourceID, &emailHost, &emailPort)
+		if err != nil {
+			log.DefaultLogger.Error("GetSettings: rows.Scan(): ", err.Error())
+			return nil, err
+		}
+
+		return &Settings{GrafanaUsername: grafanaUsername, GrafanaPassword: grafanaPassword, Email: email, EmailPassword: emailPassword, DatasourceID: datasourceID, EmailPort: emailPort, EmailHost: emailHost}, nil
```

Comment:
I guess this kind of doubles as a constructor, but could also add a conventional pure "sugar" constructor too? E.g. 

```
return NewSettings(grafanaUsername, grafanaPassword, ...), nil
```